addons/sales_team/models/res_users.py

```diff
@@ -33,3 +33,7 @@ def _compute_sale_team_id(self):
             else:
                 sorted_memberships = user.crm_team_member_ids  # sorted by create date
                 user.sale_team_id = sorted_memberships[0].crm_team_id if sorted_memberships else False
+
+    def action_archive(self):
+        self.env['crm.team.member'].search([('user_id', 'in', self.ids)]).toggle_active()
```

Comment:
Would call action_archive on member also (if a user has an archived member, archiving it is going to activate the member :D )